src/main/java/gregtech/api/capability/impl/MultiblockFuelRecipeLogic.java

```diff
@@ -19,29 +16,34 @@ public MultiblockFuelRecipeLogic(RecipeMapMultiblockController tileEntity) {
         super(tileEntity);
     }
 
+    @Nonnull
     @Override
     protected int[] runOverclockingLogic(@Nonnull IRecipePropertyStorage propertyStorage, int recipeEUt, long maxVoltage, int recipeDuration, int amountOC) {
+        // no overclocking happens other than parallelization,
+        // so return the recipe's values, with EUt made positive for it to be made negative later
+        return new int[]{-recipeEUt, recipeDuration};
+    }
+
+    @Override
+    protected void modifyOverclockPre(@Nonnull int[] values, @Nonnull IRecipePropertyStorage storage) {
         // apply maintenance penalties
-        MultiblockWithDisplayBase displayBase = this.metaTileEntity instanceof MultiblockWithDisplayBase ? (MultiblockWithDisplayBase) metaTileEntity : null;
-        int numMaintenanceProblems = displayBase == null ? 0 : displayBase.getNumMaintenanceProblems();
-
-        int[] overclock = null;
-        if (displayBase != null && ConfigHolder.machines.enableMaintenance && displayBase.hasMaintenanceMechanics()) {
-            IMaintenanceHatch hatch = displayBase.getAbilities(MultiblockAbility.MAINTENANCE_HATCH).get(0);
-            double durationMultiplier = hatch.getDurationMultiplier();
-            if (durationMultiplier != 1.0) {
-                overclock = new int[]{recipeEUt * -1, (int) Math.round(recipeDuration / durationMultiplier)};
-            }
-        }
-        if (overclock == null) {
-            overclock = new int[]{recipeEUt * -1, recipeDuration};
+        Tuple<Integer, Double> maintenanceValues = getMaintenanceValues();
+
+        // duration bonus
+        if (maintenanceValues.getSecond() != 1.0) {
+            values[1] = (int) Math.round(values[1] / maintenanceValues.getSecond());
         }
+    }
 
-        overclock[1] = (int) (overclock[1] * (1 - 0.1 * numMaintenanceProblems));
+    @Override
+    protected void modifyOverclockPost(int[] overclockResults, @Nonnull IRecipePropertyStorage storage) {
```

Comment:
Should this call super like the other calls?

Reply:
See my explanation for the comments in `modifyOverclockPre`.